packages/fluentui/react-bindings/src/hooks/useCSS.ts

```diff
@@ -0,0 +1,94 @@
+import { serializeStyles } from '@emotion/serialize';
+import { ICSSInJSStyle, ThemePrepared } from '@fluentui/styles';
+import cx from 'classnames';
+// @ts-ignore
+import _Stylis from 'stylis';
+// @ts-ignore No typings :(
+import focusVisiblePlugin from '@quid/stylis-plugin-focus-visible';
+// @ts-ignore No typings :(
+import rtlPlugin from 'stylis-plugin-rtl';
+
+import { useFluentContext } from '../context';
+
+//
+// Types
+//
+
+export type UseCSSStyle = Omit<ICSSInJSStyle, 'animationName'>;
+export type UseCSSStyleInput = string | UseCSSStyle | ((theme: ThemePrepared) => UseCSSStyle);
+
+//
+// Definitions
+//
+
+const SPECIFICITY_CLASSNAME = 'fcss';
+const CSS_PREFIX = 'f';
+
+// `stylis@3` is a CJS library, there are known issues with them:
+// https://github.com/rollup/rollup/issues/1267#issuecomment-446681320
+const Stylis = (_Stylis as any).default || _Stylis;
+const stylisOptions = {
+  cascade: true,
+  compress: false,
+  global: false,
+  keyframe: false,
+  preserve: false,
+  semicolon: false,
+};
+
+// Two separate instances are requires as we can't call Stylis plugins conditionally
+const stylis = new Stylis(stylisOptions);
+stylis.use(focusVisiblePlugin);
+
+const rtlStylis = new Stylis(stylisOptions);
+rtlStylis.use(focusVisiblePlugin);
+rtlStylis.use(rtlPlugin);
+
+// Stores a mapping between hashed string and cssified styles object
+const stylesCache: Record<string, string> = {};
+
+//
+// Hook
+//
+
+export function useCSS(...styles: UseCSSStyleInput[]) {
+  const { theme, renderer, rtl } = useFluentContext();
+
+  if (styles.length === 0) {
+    return '';
+  }
+
+  // This resolution allows to map passed classnames to actual styles that can be used on next styles
+  const resolvedStyles = styles.reduce<UseCSSStyleInput[]>((acc, style) => {
+    if (typeof style === 'string') {
+      style.split(' ').forEach(className => {
+        if (stylesCache[className] !== undefined) {
+          acc.push(stylesCache[className]);
+        }
+      });
+    } else {
+      acc.push(style);
+    }
+
+    return acc;
+  }, []);
+
+  // serializeStyles() will concat all passed styles and will resolve functions
+  const serializedStyles = serializeStyles(resolvedStyles, stylesCache, theme);
+  // ".name" is not a valid CSS classname
```

Comment:
Why? And why adding an "f" before it makes it valid?

Reply:
Changed to `// ".name" is not a valid CSS classname as it can start from a digit`, now it should be cleaner 🐱